miden-tx/tests/p2id_script_test.rs

```diff
@@ -235,13 +236,83 @@ fn test_p2id_script_multiple_assets() {
         .collect::<Vec<_>>();
 
     // Execute the transaction and get the witness
-    let transaction_result_2 = executor_2.execute_transaction(
+    let executed_transaction_2 = executor_2.execute_transaction(
         malicious_account_id,
         block_ref,
         &note_origins,
         Some(tx_script_malicious),
     );
 
     // Check that we got the expected result - TransactionExecutorError
-    assert!(transaction_result_2.is_err());
+    assert!(executed_transaction_2.is_err());
+}
+
+#[test]
+fn test_p2id_prove_and_verify() {
+    // Create assets
+    let faucet_id = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let fungible_asset: Asset = FungibleAsset::new(faucet_id, 100).unwrap().into();
+
+    // Create sender and target account
+    let sender_account_id = AccountId::try_from(ACCOUNT_ID_SENDER).unwrap();
+
+    let target_account_id =
+        AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN).unwrap();
+    let (target_pub_key, target_sk_pk_felt) = get_new_key_pair_with_advice_map();
+    let target_account =
+        get_account_with_default_account_code(target_account_id, target_pub_key, None);
+
+    // Create the note
+    let note = create_p2id_note(
+        sender_account_id,
+        target_account_id,
+        vec![fungible_asset],
+        RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]),
+    )
+    .unwrap();
+
+    // CONSTRUCT AND EXECUTE TX (Success)
+    // --------------------------------------------------------------------------------------------
+    let data_store =
+        MockDataStore::with_existing(Some(target_account.clone()), Some(vec![note.clone()]));
+
+    let mut executor = TransactionExecutor::new(data_store.clone());
+    executor.load_account(target_account_id).unwrap();
+
+    let block_ref = data_store.block_header.block_num();
+    let note_ids = data_store.notes.iter().map(|note| note.id()).collect::<Vec<_>>();
+
+    let tx_script_code = ProgramAst::parse(
+        "
+            use.miden::contracts::auth::basic->auth_tx
+    
+            begin
+                call.auth_tx::auth_tx_rpo_falcon512
+            end
+            ",
+    )
+    .unwrap();
+
+    let tx_script_target = executor
+        .compile_tx_script(
+            tx_script_code.clone(),
+            vec![(target_pub_key, target_sk_pk_felt)],
+            vec![],
+        )
+        .unwrap();
+
+    // Execute the transaction and get the witness
+    let executed_transaction = executor
+        .execute_transaction(target_account_id, block_ref, &note_ids, Some(tx_script_target))
+        .unwrap();
+
+    // Prove the transaction
+    let proof_options = ProvingOptions::default();
+    let prover = TransactionProver::new(proof_options);
+    let proven_transaction = prover.prove_transaction(executed_transaction.clone()).unwrap();
+
+    // Verify that the generated proof is valid
+    let verifier = TransactionVerifier::new(96);
```

Comment:
Let's create a constant for `96` as well - e.g., `MIN_PROOF_SECURITY_LEVEL`.

Reply:
https://github.com/0xPolygonMiden/miden-base/pull/438#discussion_r1467481737